akka-http/src/main/scala/akka/http/scaladsl/server/RejectionHandler.scala

```diff
@@ -90,12 +107,28 @@ object RejectionHandler {
       new BuiltRejectionHandler(cases.result(), notFound, isDefault)
   }
 
-  private sealed abstract class Handler
-  private final case class CaseHandler(pf: PartialFunction[Rejection, Route]) extends Handler
+  private sealed abstract class Handler {
+    def mapResponse(map: HttpResponse ⇒ HttpResponse)(implicit ec: ExecutionContext): Handler
+  }
+  private final case class CaseHandler(pf: PartialFunction[Rejection, Route]) extends Handler {
+    override def mapResponse(map: HttpResponse ⇒ HttpResponse)(implicit ec: ExecutionContext): CaseHandler = {
+      copy(pf.andThen(r ⇒ r.andThen(f ⇒ f.collect {
```

Comment:
Was clear, don't worry - applied it here https://github.com/akka/akka-http/pull/674/files#diff-d7e5a6518adba7c2e03a870299073a98R27 
Lemme see if it can be used here too